### ISSUE TITLE
Design MLModule Trainer

### ISSUE THREAD

**Issue by smassonnet** [LSIR/mlmodule#71]
MLModule trainer is a new MLModule runner that trains or fine-tune existing models.

Look at [PyTorch Ignite](https://pytorch.org/ignite/index.html) or [PyTorch Lignthning](https://www.pytorchlightning.ai/) for inspiration.

We need to define how the following logic is handled:

- Loss function definition
- Dataset with labels
- Training options (# of epocs, early stopping...)

We also need to decide which model should be made trainable.
